src/webui/btjson.cpp

```diff
@@ -887,3 +897,47 @@ QVariantMap generateSyncData(int acceptedResponseId, QVariantMap data, QVariantM
 
     return syncData;
 }
+
+QByteArray btjson::getLog(bool normal, bool info, bool warning, bool critical, int minId)
+{
+    Logger* const logger = Logger::instance();
+    QVariantList msgList;
+
+    foreach (const Log::Msg& msg, logger->getMessages()) {
```

Comment:
Don't make your life difficult. getMessages() can takes an optional `id` parameter. Pass it here and filter only the types yourself.
PS: `-1` as id value returns all messages in the log.